### ISSUE TITLE
Why not HMT by default? (instead of HABTM)

### ISSUE THREAD

**Issue by Silex** [be9/acl9#96]
Hello,

Given nowadays rails best practices, why not switch to HMT by default (instead of HABTM)?

**Comment by smathy** [comment]
Not sure what best practices you're talking about, habtm has its place for simple associations requiring no additional data in the join table. What advantage do you think using hmt would give to *this* project?

**Comment by Silex** [comment]
Well, even the `acl9` wiki (https://github.com/be9/acl9/wiki/Tutorial%3A-linking-object-and-subject-with-HMT) links to a page illustrating the various problems with HABTM: https://coderwall.com/p/752fjw/hmt-associations-with-acl9

**Comment by smathy** [comment]
I agree with you btw, that as soon as you need something else in your join table then you will need to move away from habtm, and in fact I've done that myself in a number of projects. I loved how simple both acl9 and Rails made that. All they care about are the `has_*_many` methods being there and they just work.

Now, I also have a bunch of projects where I've never changed from the habtm because actually the only purpose of the join table was for authorization. If acl9 was as you're proposing then I would have had to think of a name for the join association, it might have changed how I viewed that association, I would have ended up with a model that I didn't really need, and an additional association with the join table that I never used.

That is a much less appealing situation as a gem author than the alternative - where people who find that they do need a hmt just create it.

So, in summary, the reason I don't start with hmt is that habtm gives us everything we need and nothing we don't. There's no part of authorization that requires any other fields in the join table, so no reason for me to add them. It's easy to change if you need to, there's no new model that needs to be explained, there's no additional code required for naming the join model/table, it's just simpler this way.

Also, that article does not really illustrate problems with habtm, it just talks about the author's dislike of it and the way to use hmt with acl9 when the situation calls for it.

Thanks for your comments, I'm going to close this issue now, but feel free to reopen it if you have additional points that you didn't raise yet.

**Comment by Silex** [comment]
Alright, I can see the argument from a gem maintainer POV.

Thanks for the detailed answer.

**Comment by smathy** [comment]
:+1:
